### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by VadimKoziachiy**
- added pagination for article in UserController and ArticleController.

### CODE REVIEW

File: src/main/java/news/blog/com/controller/ArticleController.java [hunk 1]
```diff
@@ -20,9 +23,10 @@
     private final ArticleService articleService;
 
     @GetMapping
-    public ResponseEntity<Collection<ArticleDto>> getArticles()
+    public ResponseEntity<Collection<ArticleDto>> getArticles(
+            @PageableDefault(sort = {"id"}, direction = Sort.Direction.DESC) Pageable pageable)
```
Comment (alxanko) [inline src/main/java/news/blog/com/controller/ArticleController.java:28]: what are this annotation params?

File: src/main/java/news/blog/com/controller/ArticleController.java [hunk 2]
```diff
@@ -37,4 +42,16 @@ public ResponseEntity saveArticle(@RequestBody ArticleDto articleDto)
...
     }
+
+    @GetMapping("/tags")
+    public ResponseEntity<Collection<ArticleTagsResponseDto>> getTags()
+    {
+        return ResponseEntity.ok(articleService.getArticleTags());
+    }
+
+    @GetMapping("/tags/{tag}")
+    public ResponseEntity<Collection<ArticleDto>> getArticleByTag(@PathVariable String tag)
```
Comment (alxanko) [inline src/main/java/news/blog/com/controller/ArticleController.java]: it's better to use `tag` param as `RequestParam`

File: src/main/java/news/blog/com/repository/CommentRepository.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package news.blog.com.repository;
+
+import news.blog.com.model.CommentEntity;
+import org.springframework.data.repository.CrudRepository;
+
+import java.util.List;
+
+public interface CommentRepository extends CrudRepository<CommentEntity, Long>
+{
+    List<CommentEntity> findByArticle_Id(Long id);
```
Comment (alxanko) [inline src/main/java/news/blog/com/repository/CommentRepository.java]: this method can be declared w/o underscore

File: src/main/java/news/blog/com/service/impl/ArticleServiceImpl.java [hunk 4]
```diff
@@ -24,16 +27,44 @@
...
+    @Override
+    public Collection<ArticleTagsResponseDto> getArticleTags() {
+        Collection<ArticleTagsResponseDto> articleTags = new ArrayList<>();
+        Collection<ArticleEntity> articles = articleRepository.findAll();
+        Set<String> tags = articles.stream()
+                                   .map(ArticleEntity::getTag)
+                                   .collect(Collectors.toSet());
+        tags.forEach(tag ->
+                {
+                   Long quantity = articles.stream()
```
Comment (alxanko) [inline src/main/java/news/blog/com/service/impl/ArticleServiceImpl.java]: it's a complex operation
let's get unique tags directly from repositrory
and then we will count tags again with repository

File: src/main/resources/migrations/article/init-table.xml [hunk 5]
```diff
@@ -11,7 +11,7 @@
             <column name="id" type="bigint" autoIncrement="true">
                 <constraints primaryKey="true" nullable="false" unique="true"/>
             </column>
-            <column name="tags" type="varchar"/>
+            <column name="tag" type="varchar"/>
```
Comment (alxanko) [inline src/main/resources/migrations/article/init-table.xml:14]: stop changing existing migrations!
bad practice

Comment (VadimKoziachiy) [reply src/main/resources/migrations/article/init-table.xml:14]: will be done:)

File: src/main/java/news/blog/com/controller/ArticleController.java [hunk 6]
```diff
@@ -37,8 +37,8 @@ public ResponseEntity saveArticle(@RequestBody ArticleDto articleDto)
         return ResponseEntity.ok().build();
     }
 
-    @GetMapping("/{id}")
-    public ResponseEntity<ArticleDto> getArticle(@PathVariable Long id)
+    @GetMapping("/article")
```
Comment (alxanko) [inline src/main/java/news/blog/com/controller/ArticleController.java:40]: you can unite `getArticle` & `getArticleByTag` controller methods

File: src/main/java/news/blog/com/controller/CommentController.java [hunk 7]
```diff
@@ -14,14 +14,14 @@
 {
     private final CommentService commentService;
 
-    @GetMapping("/{id}")
-    public ResponseEntity getCommentsByArticle(@PathVariable Long id)
+    @GetMapping
```
Comment (alxanko) [inline src/main/java/news/blog/com/controller/CommentController.java:17]: it is not proper to do this

File: src/main/java/news/blog/com/controller/CommentController.java [hunk 8]
```diff
@@ -14,14 +14,14 @@
...
+    public ResponseEntity getCommentsByArticle(@RequestParam Long id)
     {
         return ResponseEntity.ok(commentService.getCommentsByArticle(id));
     }
 
-    @PostMapping("/{id}")
-    public ResponseEntity saveCommentsByArticle(@RequestBody CommentDto commentDto, @PathVariable Long id)
+    @PostMapping
```
Comment (alxanko) [inline src/main/java/news/blog/com/controller/CommentController.java:23]: it is not proper to do this

File: src/main/java/news/blog/com/repository/ArticleRepository.java [hunk 9]
```diff
@@ -4,17 +4,26 @@
...
+import java.util.Set;
 
 public interface ArticleRepository extends CrudRepository<ArticleEntity, Long>
 {
     List<ArticleEntity> findAll();
 
     List<ArticleEntity> findByTag(String string);
 
+    @Query(value = "SELECT tag FROM ArticleEntity ORDER BY tag ASC")
+    Set<String> findAllTags();
```
Comment (alxanko) [inline src/main/java/news/blog/com/repository/ArticleRepository.java:22]: let's read about `distinct`

**Pull request merged**